src/wazuh_db/wdb_fim.c

```diff
@@ -544,63 +544,64 @@ int wdb_fim_insert_entry2(wdb_t * wdb, const cJSON * data) {
         os_strdup(path, full_path);
         item_type = "registry_key";
     } else if (strncmp(item_type, "registry_", 9) == 0) {
-        int full_path_length;
-        char *path_escaped_slahes;
-        char *path_escaped;
-
-        if (!cJSON_IsNumber(version)) {
-            // Synchronization messages without the "version" attribute are ignored, but won't trigger any error
-            // message.
-            return 0;
-        }
 
-        path_escaped_slahes = wstr_replace(path, "\\", "\\\\");
-        path_escaped = wstr_replace(path_escaped_slahes, ":", "\\:");
-        os_free(path_escaped_slahes);
         arch = cJSON_GetStringValue(cJSON_GetObjectItem(data, "arch"));
+        value_name = cJSON_GetStringValue(cJSON_GetObjectItem(data, "value_name"));
 
-        if (arch == NULL) {
-            merror("DB(%s) fim/save registry request with no arch argument.", wdb->id);
-            os_free(path_escaped);
-            return -1;
-        }
-
-        if (strcmp(item_type + 9, "key") == 0) {
-            value_name = NULL;
-            full_path_length = snprintf(NULL, 0, "%s %s", arch, path_escaped);
+        if (version->valuedouble == 2.0) {
```

Comment:
Before accessing the `valuedouble` field, we should first check if it's a number or not. We should add the following:
```c
       if (!cJSON_IsNumber(version)) {
            // Synchronization messages without the "version" attribute are ignored, but won't trigger any error
            // message.
            return 0;
        }
```

Reply:
Done